grammar.js

```diff
@@ -1197,7 +1218,24 @@ module.exports = grammar({
       seq(
         caseInsensitive('default'),
         '(', caseInsensitive('none'), ')'
-      )
+      ),
+      seq(
+        caseInsensitive('reduce'),
+        '(', $.binary_op, ':', commaSep1($.identifier), ')',
+      ),
+    ),
+    binary_op: $ => choice(
+      '+',
+      '*',
+      caseInsensitive('\\.and\\.'),
```

Comment:
Do you think it's worth extracting out a few of these into their own unnamed expressions to avoid duplicating them in the grammar?

Reply:
I was actually wondering about making it looser, a regex like: `/\\.\w+\\.|\w+/i` (or whatever is actually correct there)

The advantage being invalid operators wouldn't be a syntax error, allowing compiler extensions and preprocessor macros.

Reply:
`\w+` doesn't seem to cause an issue